pkg/statefulset/statefulset_util.go

```diff
@@ -121,14 +120,6 @@ func CreateOrUpdateStatefulset(ctx context.Context, getUpdateCreator kubernetesC
 		log.Debug("Created StatefulSet")
 		return statefulSetToCreate, nil
 	}
-	// preserve existing certificate hash if new one is not statefulSetToCreate
-	existingCertHash, okExisting := existingStatefulSet.Spec.Template.Annotations[certs.CertHashAnnotationKey]
```

Comment:
Could you please check other usages of that cert hash key? 
e.g. [here](https://github.com/mongodb/mongodb-kubernetes/blob/b90c986cda17f63c604ac950dd8b4d64d706a4dd/controllers/operator/mongodbmultireplicaset_controller.go#L744)

The code actually looks like that it was triggerring rolling restart on cert change because we're changing podTemplate